packages/synapse-react-client/src/components/SynapseTable/SynapseTable.tsx

```diff
@@ -224,86 +266,16 @@ export function SynapseTable(props: SynapseTableProps) {
-                        colSpan={header.colSpan}
-                        style={{ width: header.getSize() }}
-                      >
-                        {header.isPlaceholder
-                          ? null
-                          : flexRender(
-                              header.column.columnDef.header,
-                              header.getContext(),
-                            )}
-                        {header.column.getCanResize() && (
-                          <div
-                            className={`resizer ${
-                              header.column.getIsResizing() ? 'isResizing' : ''
-                            }`}
-                            onMouseDown={header.getResizeHandler()}
-                            onTouchStart={header.getResizeHandler()}
-                          />
-                        )}
-                      </th>
-                    ))}
-                  </tr>
-                )
-              })}
-            </thead>
-            <tbody>
-              {table.getRowModel().rows.map(row => (
-                <tr key={row.id}>
-                  {row.getVisibleCells().map(cell => {
-                    const selectColumn = selectColumns.find(
-                      cm => cm.name === cell.column.id,
-                    )
-
-                    const shouldWrapInExpandable =
-                      selectColumn &&
-                      selectColumn.columnType !==
-                        ColumnTypeEnum.JSON /* JSON handles its own overflow*/
-                    const TableDataCellElement = shouldWrapInExpandable
-                      ? ExpandableTableDataCell
-                      : 'td'
-
-                    const renderPlaceholder = !dataHasBeenPrefetched
-
-                    return (
-                      <TableDataCellElement
-                        key={cell.id}
-                        style={{
-                          width: cell.column.getSize(),
-                          textAlign: cell.column.columnDef.meta?.textAlign,
-                        }}
-                      >
-                        {renderPlaceholder ? (
-                          <p>
-                            <Skeleton width={'80%'} height={'20px'} />
-                          </p>
-                        ) : (
-                          flexRender(
-                            cell.column.columnDef.cell,
-                            cell.getContext(),
-                          )
-                        )}
-                      </TableDataCellElement>
-                    )
-                  })}
-                </tr>
-              ))}
-            </tbody>
-          </table>
-        </div>
+        <StyledTanStackTable
+          styledTableContainerProps={{
+            className: 'SynapseTable',
+            ['data-testid']: 'SynapseTable',
+            density: 'default',
+          }}
+          table={table}
+          cellRenderer={TableCellRenderer}
```

Comment:
The table cell renderer in this case has some custom logic. Pass it as a component here